Failing to build against 0.51.0

```c
 > g++ -shared -Wall -fPIC --no-gnu-unique ./main.cpp ./workspaceLayout.cpp -g  -DWLR_USE_UNSTABLE `pkg-config --cflags pixman-1 libdrm hyprland` -std=c++23 -o workspaceLayoutPlugin.so
 > ./workspaceLayout.cpp: In member function 'virtual void CWorkspaceLayout::onEnable()':
 > ./workspaceLayout.cpp:37:24: error: cannot convert 'Hyprutils::Memory::CWeakPointer<CWorkspace>' to 'PHLWORKSPACE' {aka 'Hyprutils::Memory::CSharedPointer<CWorkspace>'}
 >    37 |         setupWorkspace(wsp);
 >       |                        ^~~
 >       |                        |
 >       |                        Hyprutils::Memory::CWeakPointer<CWorkspace>
 > ./workspaceLayout.cpp:20:52: note: initializing argument 1 of 'void CWorkspaceLayout::setupWorkspace(PHLWORKSPACE)'
 >    20 | void CWorkspaceLayout::setupWorkspace(PHLWORKSPACE pWorkspace) {
 >       |                                       ~~~~~~~~~~~~~^~~~~~~~~~
 > make: *** [Makefile:12: workspaceLayoutPlugin.so] Error 1
 For full logs, run:
   nix log /nix/store/hq57j7pf6d79ddvqfvrvlf4panv0yk00-hyprWorkspaceLayouts-0-unstable-2025-05-10.drv
```

brb forking, wonder if this is trivial or not

--- a/README.md
+++ b/README.md
@@ -19,6 +19,29 @@ plugins = [
 
 Then update via the `hyprload,update` dispatcher.
 
+## NixOS Installation
+
+Add this flake to your inputs, note that using a mismatched/unsupported Hyprland release will fail to build or load the plugin:
+
+```nix
+inputs = {
+    hyprland.url = "github:hyprwm/Hyprland"; # follows development branch of hyprland
+    hyprWorkspaceLayouts = {
+        url = "github:zakk4223/hyprWorkspaceLayouts";
+        inputs.hyprland.follows = "hyprland"; # to make sure that the plugin is built for the correct version of hyprland
+    };
+};
+```
+
+Then, in your `home-manager` config, add the plugin:
+
+```nix
+wayland.windowManager.hyprland = {
+    enable = true;
+    plugins = [ inputs.hyprWorkspaceLayouts.packages.${pkgs.system}.default ];
+};
+```
+
 ## Manual installation
 
 1. Build the plugin 

--- a/flake.nix
+++ b/flake.nix
@@ -4,7 +4,7 @@
   inputs = {
     nixpkgs.follows = "hyprland/nixpkgs";
     systems.follows = "hyprland/systems";
-    hyprland.url = "github:hyprwm/Hyprland";
+    hyprland.url = "github:hyprwm/Hyprland/v0.51.0";
   };
 
   outputs = {

--- a/flake.nix
+++ b/flake.nix
@@ -1,0 +1,64 @@
+{
+  description = "A plugin to use different layouts on workspaces";
+
+  inputs = {
+    nixpkgs.follows = "hyprland/nixpkgs";
+    systems.follows = "hyprland/systems";
+    hyprland.url = "github:hyprwm/Hyprland";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    ...
+  } @ inputs: let
+    systems = ["x86_64-linux"];
+    eachSystem = nixpkgs.lib.genAttrs systems;
+    pkgsFor = nixpkgs.legacyPackages;
+  in {
+    packages = eachSystem (system: {
+      default = self.packages.${system}.hyprWorkspaceLayouts;
+      hyprWorkspaceLayouts = pkgsFor.${system}.callPackage ({
+        lib,
+        hyprlandPlugins,
+      }:
+        hyprlandPlugins.mkHyprlandPlugin inputs.hyprland.packages.${system}.hyprland {
+          pluginName = "hyprWorkspaceLayouts";
+          version = "0-unstable-2025-09-15";
+
+          src = ./.;
+
+          nativeBuildInputs = [pkgsFor.${system}.gnumake];
+
+          buildPhase = ''
+            runHook preBuild
+            make all
+            runHook postBuild
+          '';
+
+          installPhase = ''
+            runHook preInstall
+            mkdir -p $out/lib
+            mv workspaceLayoutPlugin.so $out/lib/libhyprWorkspaceLayouts.so
+            runHook postInstall
+          '';
+
+          meta = {
+            homepage = "https://github.com/zakk4223/hyprWorkspaceLayouts";
+            description = "Workspace-specific window layouts for Hyprland";
+            license = lib.licenses.bsd3;
+            platforms = lib.platforms.linux;
+            maintainers = [];
+          };
+        }) {};
+    });
+
+    devShells = eachSystem (system: {
+      default = pkgsFor.${system}.mkShell.override {stdenv = pkgsFor.${system}.gcc14Stdenv;} {
+        name = "hyprWorkspaceLayouts";
+        buildInputs = [inputs.hyprland.packages.${system}.hyprland];
+        inputsFrom = [inputs.hyprland.packages.${system}.hyprland];
+      };
+    });
+  };
+}

--- a/flake.nix
+++ b/flake.nix
@@ -2,8 +2,7 @@
   description = "A plugin to use different layouts on workspaces";
 
   inputs = {
-    nixpkgs.follows = "hyprland/nixpkgs";
-    systems.follows = "hyprland/systems";
+    nixpkgs.url = "github:nixos/nixpkgs?ref=nixos-unstable";
     hyprland.url = "github:hyprwm/Hyprland/v0.51.0";
   };
 
@@ -18,28 +17,33 @@
   in {
     packages = eachSystem (system: {
       default = self.packages.${system}.hyprWorkspaceLayouts;
-      hyprWorkspaceLayouts = pkgsFor.${system}.callPackage ({
-        lib,
-        hyprlandPlugins,
-      }:
-        hyprlandPlugins.mkHyprlandPlugin inputs.hyprland.packages.${system}.hyprland {
-          pluginName = "hyprWorkspaceLayouts";
-          version = "0-unstable-2025-09-15";
+      hyprWorkspaceLayouts = let
+        inherit (inputs.hyprland.packages.${system}) hyprland;
+        inherit (pkgsFor.${system}) stdenvNoCC gcc14 lib;
 
+        name = "hyprWorkspaceLayouts";
+      in
+        stdenvNoCC.mkDerivation {
+          inherit name;
+          pname = name;
+          version = "0-unstable-2025-09-15";
           src = ./.;
 
-          nativeBuildInputs = [pkgsFor.${system}.gnumake];
+          inherit (hyprland) buildInputs;
+          nativeBuildInputs = hyprland.nativeBuildInputs ++ [hyprland gcc14];
+          enableParallelBuilding = true;
 
-          buildPhase = ''
-            runHook preBuild
-            make all
-            runHook postBuild
-          '';
+          dontUseCmakeConfigure = true;
+          dontUseMesonConfigure = true;
+          dontUseNinjaBuild = true;
+          dontUseNinjaInstall = true;
 
           installPhase = ''
             runHook preInstall
-            mkdir -p $out/lib
-            mv workspaceLayoutPlugin.so $out/lib/libhyprWorkspaceLayouts.so
+
+            mkdir -p "$out/lib"
+            mv workspaceLayoutPlugin.so "$out/lib/lib${name}.so"
+
             runHook postInstall
           '';
 
@@ -50,11 +54,11 @@
             platforms = lib.platforms.linux;
             maintainers = [];
           };
-        }) {};
+        };
     });
 
     devShells = eachSystem (system: {
-      default = pkgsFor.${system}.mkShell.override {stdenv = pkgsFor.${system}.gcc14Stdenv;} {
+      default = pkgsFor.${system}.mkShell {
         name = "hyprWorkspaceLayouts";
         buildInputs = [inputs.hyprland.packages.${system}.hyprland];
         inputsFrom = [inputs.hyprland.packages.${system}.hyprland];

--- a/main.cpp
+++ b/main.cpp
@@ -24,7 +24,7 @@ namespace {
     }
 
     void WSWorkspaceCreated(PHLWORKSPACE pWorkspace) {
-        g_pWorkspaceLayout->setupWorkspace(pWorkspace);
+        g_pWorkspaceLayout->setupWorkspace(PHLWORKSPACEREF{pWorkspace});
     }
 
     void WSLayoutsChanged() {

--- a/workspaceLayout.cpp
+++ b/workspaceLayout.cpp
@@ -17,7 +17,8 @@ SWorkspaceLayoutWindowData* CWorkspaceLayout::getDataFromWindow(PHLWINDOW pWindo
     return WINDOWDATA;
 }
 
-void CWorkspaceLayout::setupWorkspace(PHLWORKSPACE pWorkspace) {
+void CWorkspaceLayout::setupWorkspace(PHLWORKSPACEREF pWorkspaceRef) {
+    auto pWorkspace = pWorkspaceRef.lock();
     if (!pWorkspace) {
         //??
         return;
@@ -34,7 +35,7 @@ void CWorkspaceLayout::setupWorkspace(PHLWORKSPACE pWorkspace) {
 void CWorkspaceLayout::onEnable() {
 
     for (auto& wsp : g_pCompositor->getWorkspaces()) {
-        setupWorkspace(wsp.lock());
+        setupWorkspace(wsp);
     }
 }
 
@@ -526,7 +527,7 @@ void CWorkspaceLayout::setLayoutForWorkspace(IHyprLayout* layout, const int& ws,
         if (w.workspaceID == ws) {
             if (w.layout) {
                 if (w.layout == layout)
-                  return;
+                    return;
                 for (auto& win : g_pCompositor->m_windows) {
                     if ((win->workspaceID() != ws) || !win->m_isMapped || win->isHidden())
                         continue;

--- a/workspaceLayout.hpp
+++ b/workspaceLayout.hpp
@@ -72,7 +72,7 @@ class CWorkspaceLayout : public IHyprLayout {
 
     void                             setDefaultLayout(std::string name);
     void                             clearLayoutMaps();
-    void                             setupWorkspace(PHLWORKSPACE pWorkspace);
+    void                             setupWorkspace(PHLWORKSPACEREF pWorkspaceRef);
     void                             setupLayoutList();
 
   private:

